graphql/version_resolver.go

```diff
@@ -329,20 +329,11 @@ func (r *versionResolver) Tasks(ctx context.Context, obj *restModel.APIVersion,
 
 // TaskStatuses is the resolver for the taskStatuses field.
 func (r *versionResolver) TaskStatuses(ctx context.Context, obj *restModel.APIVersion) ([]string, error) {
-	defaultSort := []task.TasksSortOrder{
-		{Key: task.DisplayNameKey, Order: 1},
-	}
-	opts := task.GetTasksByVersionOptions{
-		Sorts:                          defaultSort,
-		IncludeBaseTasks:               false,
-		FieldsToProject:                []string{task.DisplayStatusKey},
-		IncludeBuildVariantDisplayName: false,
-	}
-	tasks, _, err := task.GetTasksByVersion(ctx, *obj.Id, opts)
+	statuses, err := task.GetTaskStatusesByVersion(ctx, *obj.Id)
 	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting version tasks: %s", err.Error()))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting task statuses for version with id '%s': %s", *obj.Id, err.Error()))
 	}
-	return getAllTaskStatuses(tasks), nil
+	return statuses, nil
```

Comment:
Out of curiosity, do we just want the unique statuses to be returned here? utility.UniqueStrings can do this

Reply:
Yes we do! I handled this in the aggregation in the `$addToSet` stage.